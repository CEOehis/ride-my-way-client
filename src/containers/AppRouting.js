import React, { Fragment } from 'react';
import {
  BrowserRouter as Router,
  Route,
} from 'react-router-dom';
import LandingPage from './pages/LandingPage';
import LoginPage from './pages/LoginPage';
import Footer from '../components/Footer';
import ProtectedRoute from './ProtectedRoute';
import SignUpPage from './pages/SignUpPage';
import HomePage from './pages/HomePage';
import RideDetailsPage from './pages/RideDetailsPage';
import OfferRidePage from './pages/OfferRidePage';

const AppRouting = () => (
  <Router>
    <Fragment>
      <Route exact path="/" component={LandingPage} />
      <Route path="/login" component={LoginPage} />
      <Route path="/register" component={SignUpPage} />
      <ProtectedRoute path="/home" component={HomePage} />
      <ProtectedRoute path="/ride/:rideId" component={RideDetailsPage} />
      <ProtectedRoute path="/create" component={OfferRidePage} />
      <Footer />
    </Fragment>
  </Router>
);

export default AppRouting;
