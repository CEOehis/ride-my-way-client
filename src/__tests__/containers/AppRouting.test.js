import React from 'react';
import ReactDOM from 'react-dom';
import AppRouting from '../../containers/AppRouting';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AppRouting />, div);
  ReactDOM.unmountComponentAtNode(div);
});
