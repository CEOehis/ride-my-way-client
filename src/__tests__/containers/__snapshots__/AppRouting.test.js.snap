// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form submit button component should render correctly 1`] = `
<BrowserRouter>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/login"
  />
  <Route
    component={[Function]}
    path="/register"
  />
  <Connect(ProtectedRoute)
    component={[Function]}
    path="/home"
  />
  <Footer />
</BrowserRouter>
`;
